cranelift/filetests/filetests/isa/x64/branches.clif

```diff
@@ -0,0 +1,88 @@
+test compile
+target x86_64
+feature "experimental_x64"
+
+function %f0(i32, i32) -> i32 {
+block0(v0: i32, v1: i32):
+  v2 = icmp eq v0, v1
+  brnz v2, block1
+  jump block2
+  ; check:  cmpl    %esi, %edi
+  ; nextln: jz      label1; j label2
+
+block1:
+  v3 = iconst.i32 1
+  ; check: movl    $$1, %eax
+  return v3
+  ; check: ret
+
+block2:
+  v4 = iconst.i32 2
+  ; check: movl    $$2, %eax
+  return v4
+  ; check: ret
+}
+
+function %f1(i32, i32) -> i32 {
+block0(v0: i32, v1: i32):
+  v2 = icmp eq v0, v1
+  brz v2, block1
+  jump block2
+  ; check:  cmpl    %esi, %edi
+  ; nextln: jnz     label1; j label2
+
+block1:
+  v3 = iconst.i32 1
+  ; check: movl    $$1, %eax
+  return v3
+  ; check: ret
+
+block2:
+  v4 = iconst.i32 2
+  ; check: movl    $$2, %eax
+  return v4
+  ; check: ret
+}
+
+function %f2(i32, i32) -> i32 {
+block0(v0: i32, v1: i32):
+  v2 = ifcmp v0, v1
+  brif eq v2, block1
+  jump block2
+  ; check:  cmpl    %esi, %edi
+  ; nextln: jz      label1; j label2
```

Comment:
Hm... so we _have_ to jump for both sides of the branch, right? I would hope this doesn't affect performance too much (it's an unconditional jump) but it basically means that the Cranelift `fallthrough` instruction does not have the meaning Cranelift users might expect (I would expect no instruction would be necessary because `fallthrough` would imply the blocks are contiguous). I would suggest we resolve this in some way: 
 - perhaps add an issue to figure out how to preserve the `fallthrough` assumptions in the new backend?
 - perhaps remove `fallthrough`, forcing users to use an unconditional jump instead?

Reply:
Here's the applicable `fallthrough` documentation: "Fall through to the next block. This is the same as `jump`, except the destination block must be the next one in the layout."

Reply:
`fallthrough` mainly exists for the old backend. The old backend automatically replaces unconditional jumps to the next block with it.

Reply:
Indeed, I think that what this patch does is semantically correct (if the code-generator follows the rules, then we will generate correct machine code), and in fact even correct for a superset of defined behavior (if the code-gen puts the blocks in the wrong order, we will still use the named target of `fallthrough`), but loses the block-ordering hint because our block-ordering logic in `BlockLoweringOrder` does its own DFS.

So I think the right thing to do here would be to (i) define `fallthrough` to be a hint, i.e., equivalent to a jump but expressing a desire to place the named block next in final code (this is compatible with current definition), and (ii) use that hint in the lowering logic. I can create an issue for this, as @abrown suggests. Thanks!